src/server/transaction.cc

```diff
@@ -157,7 +155,7 @@ void Transaction::InitShardData(absl::Span<const PerShardCache> shard_index, siz
     }
   }
 
-  CHECK(args_.size() == num_args);
+  CHECK_EQ(args_.size(), num_args);
```

Comment:
The PR title, description and the CHECK that is still here is not very coherent to me.
Consider clarifying.

Reply:
this is a different check-fail